src/Http/Http.Abstractions/src/Validation/ValidatableParameterInfo.cs

```diff
@@ -58,14 +57,17 @@ protected ValidatableParameterInfo(
     /// </remarks>
     public virtual async Task ValidateAsync(object? value, ValidateContext context, CancellationToken cancellationToken)
     {
-        Debug.Assert(context.ValidationContext is not null);
-
         // Skip validation if value is null and parameter is optional
         if (value == null && ParameterType.IsNullable())
         {
             return;
         }
 
+        // ValidationContext requires a non-null value although the invocation pattern that we use
+        // calls `GetValidationResult` and passes the value there. `GetValidationResult` tolerates
+        // null values so we only need to set a non-null value to the ValidationContext here.
+        context.ValidationContext ??= new ValidationContext(value ?? new object(), displayName: DisplayName, serviceProvider: null, items: null);
```

Comment:
Can we avoid allocating a new value each time in the case we don't need one?

Reply:
Oh yeah, like only newing up in the getter?